### ISSUE TITLE
Modifying for hypothesis testing

### ISSUE THREAD

**Issue by seandickert** [RaviSoji/plda#47]
For comparing two unseen speakers to get a score for similarity, is this as simple as:

```
score_same = model.calc_logp_marginal_likelihood(join(x, y))
score_diff = model.calc_logp_marginal_likelihood(x) * model.calc_logp_marginal_likelihood(y)
score = score_same - score_diff
```

**Comment by RaviSoji** [comment]
Not quite. The function computes the marginal likelihood for a given set of data (in U_model space) using the prior parameters (see line 120 of https://github.com/RaviSoji/plda/blob/master/plda/model.py).

It's tricky because there is one prior distribution, but many sets of posterior parameters (one for each category). One way to do this is to build a K x K matrix where K is the number of categories in your training data, where each element is a product of a pair of posterior predictives p(X = x_1 | k = i) and p(X = x_2 | k = j), and i and j are row and column indices, respectively, representing one of the K categories. You would use `model.calc_logp_posterior_predictive()` to compute the densities, and of course working in log space the products will become sums.

Without a prior on the pair being same or different, you can get a same-diff score by summing the diagonal entries of this matrix and dividing by sum of all the values in the matrix. However, without putting a 50/50 prior on same-diff, you will find this approach to perform poorly as the number of categories increases.

The way to do what you want so the score is useful is to incorporate a uniform prior on a pair of data being same or different: this is implemented by upweighting the diagonal elements (product of densities) by K^2 - K, and then perform the normalization above.

Good luck!

**Comment by seandickert** [comment]
I've submitted a PR with the hypothesis test implemented on the plda model object. Added test case
